packages/utils/src/cjs.js

```diff
@@ -1,11 +1,15 @@
 import { join } from 'path'
 
+export function isHMRCompatible (id) {
+  return !/[/\\]mongoose[/\\/](lib[/\\/])?index.js/.test(id)
```

Comment:
Why not marking entire mongoose dep as incompatible? A regex matching `node_modules/mongoose` as a part of id

Reply:
mongoose has lots of files and it does not bundle them. clearing entire deps causes Maximum call stack size exceeded. (Honestly, I don't understand why)

Reply:
Do you have a repo to quickly check? Maybe it is a nuxt HMR bug

Reply:
I'll have one, let me push it to github

Reply:
@pi0 https://github.com/farnabaz/nuxt-mongoose

Reply:
Can we create a nuxt module for mongoouse and clear cache via `watchers` or `modules:before` hook ?

Reply:
@farnabaz Typescript recursion should be fixed by (#7966)

Reply:
@clarkdo Seems nice idea. Also there is an alternative way i used once to use [unload](#7358) method. 

Since it is really minor case (serverMiddleware + HMR + mongoose) but not uncommon setup, it may worth excluding mongoose by default. Nuxt 3 will isolate functions anyway so we won't have such issues for HMR.

Reply:
@pi0 I have updated the regex based on based on fix #7966